api/utils.py

```diff
@@ -365,9 +365,22 @@ def formatEdition(
 
             editionDict['items'].append(itemDict)
 
-        editionDict['items']\
-            .sort(key=lambda x: (cls.SOURCE_PRIORITY[x['source']], cls.sortByMediaType(x['links'][0])))
-
+        emptyListFlag = False
+        
+        #This for loop is meant to check if a empty links array exists in a item dictionary
+            #If one exists, then sorting by source priority and empty/non-empty priority occurs
+            #Otherwise, sorting by source and link media type occurs
+        for itemDict in editionDict['items']:
+            if itemDict['links'] == []:
+                editionDict['items']\
+                    .sort(key=lambda x: (cls.SOURCE_PRIORITY[x['source']], x['links'] == []))
```

Comment:
This looks good but is `x['links'] == []` correct? Seems like that would sort empty lists to the top and `!=` is what we want. Not sure though, just want to check!

Reply:
That is the right way to correct sorting for empty lists since the test `test_formatEdition_v2_reader_flag_2` has a item with a empty list and a item with a non empty list that both have a gutenberg source in which the non empty list becomes the first element in the list after the formattedEdition method is called. This is the result:

`{'id': 'ed1', 'work': {}, 'publication_date': 2000, 'items': [{'id': 'it2', 'links': [{'link_id': 'li2', 'mediaType': 'application/webpub+json', 'url': 'testURI', 'flags': {'test': True}}], 'rights': [], 'physical_location': {}, 'source': 'gutenberg', 'item_id': 'it2', 'location': None}, {'id': 'it4', 'links': [], 'rights': [], 'physical_location': {}, 'source': 'gutenberg', 'item_id': 'it4', 'location': None}, {'id': 'it1', 'links': [{'link_id': 'li1', 'mediaType': 'application/epub+xml', 'url': 'testURI', 'flags': {'test': True, 'reader': False}}], 'rights': [{'source': 'test', 'license': 'testLicense', 'rightsStatement': 'testStatement'}], 'physical_location': {'name': 'test'}, 'source': 'hathitrust', 'item_id': 'it1', 'location': 'test'}], 'links': [{'link_id': 'co1', 'mediaType': 'image/png', 'url': 'testCover'}], 'edition_id': 'ed1', 'work_uuid': 'uuid1', 'inCollections': None, 'work_title': None, 'work_authors': None}`